modules/git/repo_base_gogit.go

```diff
@@ -68,10 +65,22 @@ func OpenRepositoryWithAlternates(ctx context.Context, repoPath string, altPath
 		KeepDescriptors:      true,
 		LargeObjectThreshold: setting.Git.LargeObjectThreshold,
 	}
-	if altPath == "" {
-		altPath = repoPath
+
+	// attach possible alternates
+	bs, err := os.ReadFile(filepath.Join(repoPath, "objects", "info", "alternates"))
+	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			return nil, err
+		}
+	} else {
+		scanner := bufio.NewScanner(bytes.NewReader(bs))
+		for scanner.Scan() {
+			// FIXME: we just use the first alternate because it seems go-git don't support multiple alternates yet
+			altPath := scanner.Text()
+			options.AlternatesFS = osfs.New(altPath)
```

Comment:
If `altPath` is relative to the host root dir, you probably want:
`options.AlternatesFS = osfs.New("/")`

I think this might work as well:
`options.AlternatesFS = osfs.New(altPath, osfs.WithBoundOS())`

Reply:
Thanks! Changed in 4f26e65, but it seems I'm still not lucky.
Does this work in Windows?
```go
options.AlternatesFS = osfs.New("/")
```

Reply:
My understanding is that `osfs.New("")` and `osfs.New("/")` are equivalent. But looking across `go-git`, we seem to only use the former - for example on the [default loader](https://github.com/go-git/go-git/blob/7c44bdda2fcaf50b7b648419195534b0a6c92419/plumbing/transport/server/loader.go#L14).

Reply:
FYI: `osfs.New("")` is not the same as `osfs.New("/")`

And I think there are some other problems in go-git: https://github.com/go-git/go-git/issues/1006